tic-tac-toe-the3rd/src/components/Game.js

```diff
@@ -0,0 +1,96 @@
+import React, { useContext } from 'react';
+import Cells from './Cell';
+import BoardContext from '../context/BoardContext';
+
+
+function Board(props) {
+  const { board, setBoard, turn, setTurn, history, setHistory } = useContext(BoardContext);
+
+  function declareWinner(board) {
+    const winningConditions = [
+      [0, 1, 2],
+      [3, 4, 5],
+      [6, 7, 8],
+      [0, 3, 6],
+      [1, 4, 7],
+      [2, 5, 8],
+      [0, 4, 8],
+      [2, 4, 6]
+    ];
+
+    for (let i = 0; i < winningConditions.length; i++) {
+      const [a, b, c] = winningConditions[i];
+      if (board[a] && board[a] === board[b] && board[a] === board[c]) {
+        return board[a];
+      }
+    }
+    return null;
+
+  }
+  const winner = declareWinner(board);
+  let status;
+
+  if (winner) {
+    status = `The winner is ${winner}`;
+  }
+
+  function clickHandler(num) {
+    // whenever someone clicks a button, they apply a sate to that specific button
+    // we can set this state into the boardState and then check if there's a winner
+    if (!winner && !board[num]) {
+      const updatedBoard = [...board];
+      updatedBoard[num] = turn;
+      setBoard(updatedBoard);
+      setTurn((turn === 'X') ? 'O' : 'X');
+      const recordedHistory = history.slice();
+      recordedHistory.push(updatedBoard);
+      setHistory(recordedHistory);
+    }
+  }
+
+
+  function travel(i) {
+    const boardInHistory = [...history];
+    setBoard(boardInHistory[i]);
+    setTurn((i % 2 === 0) ? 'O' : 'X');
+  }
+
+  const moves = history.map((cell, i) => {
+    const actions = i ? `Go to move ${i}` : 'Go to game start';
+    return (
+      <li key={i}>
+        <button onClick={() => travel(i)}>{actions}</button>
+      </li>
+    );
+  });
+
+  return (
+    <div className="board">
+      <h1>Player {turn}, it's your turn</h1>
+      <h3>{status}</h3>
+      <h3>{moves}</h3>
+
+      <div className="row">
+        <Cells click={() => clickHandler(0)
```

Comment:
I'd definitely recommend using `map`  here for a couple of reasons. 1). It's a little bit more expressive, as it shows that you are rendering a cell for each item in the board and 2). It would be less for you to type :-) 

```jsx
{board.map((cellValue, i) => <Cells click={() => clickHandler(i)} value={cellvalue} /> }
```

Reply:
Also, one advantage to moving the `clickHandler` function onto the context is that you wouldn't need to pass it as a prop- the `Cell` component could just pull the function it needed from the context on its own.